### PR TITLE
Ma fix pkgsigcheck

### PR DESCRIPTION
**Pull request by mlandres**
Fixing [bsc#1091624](https://bugzilla.suse.com/show_bug.cgi?id=1091624)

The packages sig check code was moved into a FileChecker passed down to the Fetcher. This avoids the package being moved into the cache before the check has completed.

### CODE REVIEW

**Comment by mlandres** [comment]
TODO:
- [ ]  Need to fix the case of using delta rpms to build the rpm localy. They must be built outside the cache, then be checked before moving them to the cache. Will add one more commit for this.

File: zypp/repo/PackageProvider.cc [hunk 1]
```diff
@@ -189,22 +275,20 @@ namespace zypp
...
-	    _retry = true;
-	    break;
 	  case repo::DownloadResolvableReport::IGNORE:
 	    WAR << _package->asUserString() << ": " << "User requested to accept insecure file" << endl;
 	    break;
 	  default:
+	  case repo::DownloadResolvableReport::RETRY:
 	  case repo::DownloadResolvableReport::ABORT:
-	    ZYPP_THROW(AbortRequestException("User requested to abort"));
+	    ZYPP_THROW(RpmSigCheckException(action_r,"UFF..."));
```
Comment (bzeller) [inline zypp/repo/PackageProvider.cc]: UFF..... ? ;D

**Comment by mlandres** [comment]
Final commit added. TODOs are done.

File: zypp/repo/PackageProvider.cc [hunk 2]
```diff
@@ -586,18 +579,28 @@ namespace zypp
...
       report()->finishDeltaApply();
 
-      return ManagedFile( destination, filesystem::unlink );
+      // Check and move it into the cache
+      // Here the rpm itself is ready. If the packages sigcheck fails, it
+      // makes no sense to return a ManagedFile() and fallback to download the
+      // full rpm. It won't be different. So let the exceptions escape...
+      rpmSigFileChecker( builddest );
+      if ( filesystem::hardlinkCopy( builddest, cachedest ) != 0 )
+	ZYPP_THROW( Exception( str::Str() << "Can't hardlink/copy " << builddest << " to " << cachedest ) );
```
Comment (bzeller) [inline zypp/repo/PackageProvider.cc]: That line should be intented

Comment (bzeller) [reply zypp/repo/PackageProvider.cc]: Actually it seems it is, but with a \t rather than spaces. The Github diff view does not show that properly.

**Pull request merged**